core/src/main/scala/org/apache/gearpump/util/Util.scala

```diff
@@ -33,6 +33,11 @@ import scala.util.Try
 
 object Util {
   val LOG = LogUtil.getLogger(getClass)
+  private val appNamePattern = "^[a-zA-Z_]+[a-zA-Z0-9_]+$".r.pattern
```

Comment:
"^[a-zA-Z_]+[a-zA-Z0-9_]+$"

should remove the first "+", otherwise it is tricky to match the regex.

 "^[a-zA-Z_][a-zA-Z0-9_]+$"